client/src/holdings/AddTradeDialog.tsx

```diff
@@ -38,7 +61,34 @@ export function AddTradeDialog({
   function handleCancel() {
     setOpen(false);
   }
-
+  async function onSubmit(data: CreateTradeData) {
```

Comment:
It's not actually of type CreateTradeData, you'll need to make a new interface IFormInput like in the file I sent.